code/datums/ai/generic/find_and_set.dm

```diff
@@ -68,11 +68,8 @@
 /datum/ai_behavior/find_and_set/in_list
 
 /datum/ai_behavior/find_and_set/in_list/search_tactic(datum/ai_controller/controller, locate_paths, search_range)
-	var/list/found = list()
-	for(var/locate_path in locate_paths)
-		var/single_locate = ..(controller, locate_path, search_range)
-		if(single_locate)
-			found += single_locate
+	var/list/look_for_list = typecacheof(locate_paths)
```

Comment:
You should make the locate_paths value a typecache rather then a list.
Typecache generation is semi expensive, only doing it once is gonna speed this up further